sia/miner/mine.go

```diff
@@ -69,27 +76,40 @@ func (m *Miner) mine() {
 	}
 }
 
-// SolveBlock grabs a block from the miner and grinds on the block, trying to
-// find a winning solution.
-//
-// SolveBlock locks the miner for long enough to grab a block, and then unlocks
-// the miner for the remaining work, which does not interact with the miner.
-func (m *Miner) SolveBlock() (b consensus.Block, solved bool, err error) {
-	// Lock the miner and grab the information necessary for grinding hashes.
-	m.mu.RLock()
-	b = m.blockForWork()
-	target := m.target
-	iterations := m.iterationsPerAttempt
-	m.mu.RUnlock()
+// solveBlock takes a block, target, and number of iterations as input and
+// tries to find a block that meets the target. This function can take a long
+// time to complete, and should not be called with a lock.
+func (m *Miner) solveBlock(blockForWork consensus.Block, target consensus.Target, iterations uint64) (b consensus.Block, solved bool, err error) {
+	// solveBlock could operate on a pointer, but it's not strictly necessary
+	// and it makes calling weirder/more opaque.
+	b = blockForWork
 
 	// Iterate through a bunch of nonces (from a random starting point) and try
 	// to find a winnning solution.
 	for maxNonce := b.Nonce + iterations; b.Nonce != maxNonce; b.Nonce++ {
 		if b.CheckTarget(target) {
+			// TODO: Eventually, we'll be calling AcceptBlock directly.
+			// Unfortunately, that throws off all of the other modules right
+			// now.
+			/*
+				// TODO: If debug, check the error value of AcceptBlock and panic
+				// for err != nil.
+				m.state.AcceptBlock(b)
+			*/
 			m.mu.RLock()
 			m.blockChan <- b
 			m.mu.RUnlock()
```

Comment:
This code is deprecated. It'll just be one line `m.state.AcceptBlock(b)` once other modules no longer depend on the core for updates.